### PR TITLE
Virtualimage funcs

### PR DESCRIPTION
**Pull request by alex-rakowski**

### CODE REVIEW

File: py4DSTEM/process/virtualimage/virtualimage.py [hunk 1]
```diff
@@ -208,7 +224,7 @@ def plot_mask_overlay(mask, dp=None, datacube=None, reduce_func=np.mean, alpha=0
 #### Virtual Imaging Functions ####
 
 ##### py4DSTEM funcs V0.13.0 ####
-def get_virtualimage_rect(datacube, geometry, verbose=True):
+def get_virtualimage_rect_old(datacube, geometry, verbose=True, *args, **kwargs):
```
Comment (bsavitzky) [inline py4DSTEM/process/virtualimage/virtualimage.py:227]: add "_" to all these `old` fns

File: py4DSTEM/process/virtualimage/virtualimage.py [hunk 2]
```diff
@@ -565,189 +581,185 @@ def get_virutalimage_rect_tensordot(datacube, geometry, spicy=False):
...
-#     return out
+    """
+    Get a virtual image from a py4DSTEM datacube object, and will operate on in memory (np.ndarray), memory mapped (np.memmap) or dask arrays (da.Array)
+    This function can be operated in two modes: 
+        - passing a detector geometry, will generate a boolean mask corresponding to detector geometry these require a tuple to define the detector geometry: 
+            - 'rect': (4-tuple) the corners (qx0,qxf,qy0,qyf)
+            - 'circ': (2-tuple) (center,radius) where center=(qx0,qy0)
+            - 'ann': (2-tuple) (center,radii) where center=(qx0,qy0) and
+                radii=(ri,ro)
+            - 'mask': (2D boolean array)
```
Comment (bsavitzky) [inline py4DSTEM/process/virtualimage/virtualimage.py:682]: rm

File: py4DSTEM/process/virtualimage/virtualimage.py [hunk 3]
```diff
@@ -565,189 +581,185 @@ def get_virutalimage_rect_tensordot(datacube, geometry, spicy=False):
...
-#         # out[rx,ry] = np.sum(arr[rx,ry][mask]) 
-#     return out
+    """
+    Get a virtual image from a py4DSTEM datacube object, and will operate on in memory (np.ndarray), memory mapped (np.memmap) or dask arrays (da.Array)
+    This function can be operated in two modes: 
+        - passing a detector geometry, will generate a boolean mask corresponding to detector geometry these require a tuple to define the detector geometry: 
+            - 'rect': (4-tuple) the corners (qx0,qxf,qy0,qyf)
+            - 'circ': (2-tuple) (center,radius) where center=(qx0,qy0)
+            - 'ann': (2-tuple) (center,radii) where center=(qx0,qy0) and
+                radii=(ri,ro)
```
Comment (bsavitzky) [inline py4DSTEM/process/virtualimage/virtualimage.py:681]: move to Args

Comment (alex-rakowski) [reply py4DSTEM/process/virtualimage/virtualimage.py:681]: D

**Comment by alex-rakowski** [comment]
All comments are resolved, and it passed all the testing and profiling I've done

**Review by bsavitzky — APPROVED** [review]

**Pull request merged**